### PR TITLE
Update CHANGE_LOG for 0.39.0

### PR DESCRIPTION
**Pull request by stuartarchibald**
As title.

### CODE REVIEW

**Comment by esc** [comment]
@stuartarchibald thank you for PR, I added the list of PRs and authors to the `CHANGE_LOG`.

File: CHANGE_LOG [hunk 1]
```diff
@@ -1,7 +1,65 @@
...
+  general use (additional work is needed).
+
+Pull-Requests:
+
+* PR `#328 <https://github.com/numba/llvmlite/pull/328>`_: Build C files separately on Linux and support parallel make (`Michał Górny <https://github.com/mgorny>`_)
+* PR `#754 <https://github.com/numba/llvmlite/pull/754>`_: manylinux2014 aarch64 wheels with system compilers (`esc <https://github.com/esc>`_)
+* PR `#760 <https://github.com/numba/llvmlite/pull/760>`_: add support for attaching metadata to global variables (`Graham Markall <https://github.com/gmarkall>`_ `John Törnblom <https://github.com/john-tornblom>`_)
+* PR `#786 <https://github.com/numba/llvmlite/pull/786>`_: Update Windows and OSX CI images. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#801 <https://github.com/numba/llvmlite/pull/801>`_: Update ffi.py (`franzhaas <https://github.com/franzhaas>`_)
+* PR `#803 <https://github.com/numba/llvmlite/pull/803>`_: llvm::Module::GetSourceFileName ()
```
Comment (gmarkall) [inline CHANGE_LOG:26]:
```suggestion
* PR `#803 <https://github.com/numba/llvmlite/pull/803>`_: llvm::Module::GetSourceFileName (`J. Aaron Pendergrass <https://github.com/japendergrass>`_)
```

File: CHANGE_LOG [hunk 2]
```diff
@@ -1,7 +1,65 @@
...
+
+Authors:
+
+* `Andrii Oriekhov <https://github.com/andriyor>`_
+* `Andreas Wrisley <https://github.com/anlofw>`_
+* `Andre Masella <https://github.com/apmasell>`_
+* `esc <https://github.com/esc>`_
+* `franzhaas <https://github.com/franzhaas>`_
+* `Graham Markall <https://github.com/gmarkall>`_
+* `John Törnblom <https://github.com/john-tornblom>`_
```
Comment (gmarkall) [inline CHANGE_LOG:56]:
```suggestion
* `J. Aaron Pendergrass <https://github.com/japendergrass>`_
* `John Törnblom <https://github.com/john-tornblom>`_
```

File: CHANGE_LOG [hunk 3]
```diff
@@ -1,7 +1,65 @@
...
+* `Michał Górny <https://github.com/mgorny>`_
+* `Stan Seibert <https://github.com/seibert>`_
+* `Siu Kwan Lam <https://github.com/sklam>`_
+* `stuartarchibald <https://github.com/stuartarchibald>`_
 
-* In development
+Total authors: 11
```
Comment (gmarkall) [inline CHANGE_LOG:62]:
```suggestion
Total authors: 12
```

**Review by gmarkall — CHANGES_REQUESTED** [review]
Looks good to me, except that J. Aaron Pendergrass was missed off the credits, probably because Github can't correlate the commits from PR #803 with the username.

**Comment by esc** [comment]
> Looks good to me, except that J. Aaron Pendergrass was missed off the credits, probably because Github can't correlate the commits from PR #803 with the username.

Probably, yes, some part of the automation failed, thank you for catching!

File: CHANGE_LOG [hunk 4]
```diff
@@ -1,7 +1,66 @@
...
+* `franzhaas <https://github.com/franzhaas>`_
+* `Graham Markall <https://github.com/gmarkall>`_
+* `J. Aaron Pendergrass <https://github.com/japendergrass>`_
+* `John Törnblom <https://github.com/john-tornblom>`_
+* `Michał Górny <https://github.com/mgorny>`_
+* `Stan Seibert <https://github.com/seibert>`_
+* `Siu Kwan Lam <https://github.com/sklam>`_
+* `stuartarchibald <https://github.com/stuartarchibald>`_
 
-* In development
```
Comment (gmarkall) [inline CHANGE_LOG:4]: Looking at previous history, it looks like the "In development" bullet point and version should be added with the next version bump PR (c.f. https://github.com/numba/llvmlite/commit/37b8e805f5698b2c3c7bc85b79e3c874b710471e) so I think this removal is fine.

**Review by gmarkall — APPROVED** [review]
Thanks for the update - this now looks good to me.

File: CHANGE_LOG (hunk 4 above)
Comment (esc) [reply CHANGE_LOG:4]: yes indeed, we will re-initialize the CHANGE_LOG post release with a new version header and this bullet as a placeholder.

File: CHANGE_LOG [hunk 5]
```diff
@@ -1,7 +1,66 @@
...
+* PR `#815 <https://github.com/numba/llvmlite/pull/815>`_: update local references (`esc <https://github.com/esc>`_)
+* PR `#816 <https://github.com/numba/llvmlite/pull/816>`_: remove configuration landscape service as it is no longer used (`esc <https://github.com/esc>`_)
+* PR `#817 <https://github.com/numba/llvmlite/pull/817>`_: remove uppper limit on Python requires (`esc <https://github.com/esc>`_)
+* PR `#819 <https://github.com/numba/llvmlite/pull/819>`_: adding rc and final release checklist templates (`esc <https://github.com/esc>`_)
+* PR `#823 <https://github.com/numba/llvmlite/pull/823>`_: Add section to globals (`Andreas Wrisley <https://github.com/anlofw>`_)
+* PR `#824 <https://github.com/numba/llvmlite/pull/824>`_: add GitHub URL for PyPi (`Andrii Oriekhov <https://github.com/andriyor>`_)
+* PR `#825 <https://github.com/numba/llvmlite/pull/825>`_: Add flag handling to more instructions. (`stuartarchibald <https://github.com/stuartarchibald>`_ `Andre Masella <https://github.com/apmasell>`_)
+* PR `#826 <https://github.com/numba/llvmlite/pull/826>`_: Deprecated `llvmlite.llvmpy` (`Andre Masella <https://github.com/apmasell>`_)
```
Comment (stuartarchibald) [inline CHANGE_LOG:37]: Should this be mentioned in the summary above? Seems quite important.

File: CHANGE_LOG [hunk 6]
```diff
@@ -1,7 +1,66 @@
-v0.39.0 (XXX YY, ZZZZ)
-----------------------
+v0.39.0 (TBC)
```
Comment (stuartarchibald) [inline CHANGE_LOG:1]: This needs an actual date before merge.

File: CHANGE_LOG [hunk 7]
```diff
@@ -1,7 +1,66 @@
...
+
+Pull-Requests:
+
+* PR `#328 <https://github.com/numba/llvmlite/pull/328>`_: Build C files separately on Linux and support parallel make (`Michał Górny <https://github.com/mgorny>`_)
+* PR `#754 <https://github.com/numba/llvmlite/pull/754>`_: manylinux2014 aarch64 wheels with system compilers (`esc <https://github.com/esc>`_)
+* PR `#760 <https://github.com/numba/llvmlite/pull/760>`_: add support for attaching metadata to global variables (`Graham Markall <https://github.com/gmarkall>`_ `John Törnblom <https://github.com/john-tornblom>`_)
+* PR `#786 <https://github.com/numba/llvmlite/pull/786>`_: Update Windows and OSX CI images. (`stuartarchibald <https://github.com/stuartarchibald>`_)
+* PR `#801 <https://github.com/numba/llvmlite/pull/801>`_: Update ffi.py (`franzhaas <https://github.com/franzhaas>`_)
+* PR `#803 <https://github.com/numba/llvmlite/pull/803>`_: llvm::Module::GetSourceFileName (`J. Aaron Pendergrass <https://github.com/japendergrass>`_)
+* PR `#807 <https://github.com/numba/llvmlite/pull/807>`_: Exclude ExecutionEngine tests on linux 32 (`esc <https://github.com/esc>`_)
```
[the remark this answers is not in the document]
Comment (esc) [reply CHANGE_LOG:28]: this was the PR to initialize 0.39.0dev0, added it now.

File: CHANGE_LOG (hunk 6 above)
Comment (stuartarchibald) [reply CHANGE_LOG]: Done in 94cb1af

File: CHANGE_LOG [hunk 8]
```diff
@@ -1,7 +1,74 @@
...
+  compatibility with other projects and improve start-up times.
+* Linux builds now use the parallel option to ``make`` to speed up building the
+  FFI library.
+* Preliminary work to expose LLVM's optimization-remarks interface has been
+  undertaken. The bindings are exposed and tested, but not yet documented for
+  general use (additional work is needed).
+
+Deprecations:
+
+* The `llvmlite.llvmpy` module has been deprecated as the functionality it
```
Comment (esc) [inline CHANGE_LOG]:
```suggestion
* The ``llvmlite.llvmpy`` module has been deprecated as the functionality it
```

File: CHANGE_LOG [hunk 9]
```diff
@@ -1,7 +1,74 @@
...
+  provides is available through the `llvmlite.ir` module. See the deprecation
```
Comment (esc) [inline CHANGE_LOG]:
```suggestion
  provides is available through the ``llvmlite.ir`` module. See the deprecation
```

**Review by esc — CHANGES_REQUESTED** [review]
The entry in the CHANGE_LOG regarding `llvmlite.llvmpy` deprecations needs additional backticks, currently it renders like:
Screen Shot 2022-06-27 at 18 05 41

**Comment by stuartarchibald** [comment]
@esc thanks, see f9c8dbc, which should fix.

**Pull request merged**